pkg/metrics/metricvalues/values.go

```diff
@@ -0,0 +1,104 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metricvalues
+
+import (
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+)
+
+// MetricValue represents a key-value pair for logging with a known key
+type MetricValue struct {
```

Comment:
Could consider just using `prometheus.Labels`? It's just a `map[string]string` - but if we ever wanted to actually turn this into a prom metric we'd need to compy with that anyway, and in your case almost everything is a string anyway except `OSDiskSizeGB`